retrofit/src/test/java/retrofit2/RequestFactoryTest.java

```diff
@@ -609,10 +609,10 @@ Call<ResponseBody> method(@HeaderMap Map<String, Object> headers) {
   }
 
   @Test
-  public void headerMapMustBeAMap() {
+  public void headerMapMustBeAMapOrHeaders() {
     class Example {
       @GET("/")
-      Call<ResponseBody> method(@HeaderMap List<String> headers) {
+      Call<ResponseBody> method(@HeaderMap okhttp3.Headers headers, @HeaderMap List<String> headerMap) {
```

Comment:
Why was the `Headers` parameter added? If we want to test it works then it should be done separately from testing a type which does not work.

Reply:
Cause I see the workable ones already exist.

https://github.com/square/retrofit/blob/296fcd962956c1c806b3bcf08b91edb96a8ada75/retrofit/src/test/java/retrofit2/RequestFactoryTest.java#L715-L737

https://github.com/square/retrofit/blob/296fcd962956c1c806b3bcf08b91edb96a8ada75/retrofit/src/test/java/retrofit2/RequestFactoryTest.java#L739-L763